nikola/plugins/command/import_wordpress.py

```diff
@@ -931,14 +954,32 @@ def import_postpage_item(self, item, wordpress_namespace, out_folder=None, attac
                     meta_slug = slug
                 tags, other_meta = self._create_metadata(status, excerpt, tags, categories,
                                                          post_name=os.path.join(out_folder, slug))
-                self.write_metadata(os.path.join(self.output_folder, out_folder,
-                                                 out_meta_filename),
-                                    title, meta_slug, post_date, description, tags, **other_meta)
-                self.write_content(
-                    os.path.join(self.output_folder,
-                                 out_folder, out_content_filename),
-                    content,
-                    rewrite_html)
+
+                meta = {
+                    "title": title,
+                    "slug": meta_slug,
+                    "date": post_date,
+                    "description": description,
+                    "tags": ','.join(tags),
+                }
+                meta.update(other_meta)
+                if self.onefile:
```

Comment:
Because compilers don't support saving in 2 files and I don't want to chase
that right now.

El mar., mar. 8, 2016 19:38, Felix Fontein <notifications@github.com>
escribió:

> In nikola/plugins/command/import_wordpress.py
> <https://github.com/getnikola/nikola/pull/2292#discussion_r55441307>:
>
> > -                                    title, meta_slug, post_date, description, tags, **other_meta)
> > -                self.write_content(
> > -                    os.path.join(self.output_folder,
> > -                                 out_folder, out_content_filename),
> > -                    content,
> > -                    rewrite_html)
> > +
> > +                meta = {
> > +                    "title": title,
> > +                    "slug": meta_slug,
> > +                    "date": post_date,
> > +                    "description": description,
> > +                    "tags": ','.join(tags),
> > +                }
> > +                meta.update(other_meta)
> > +                if self.onefile:
>
> Why not pass self.onefile as another argument to write_post, so that all
> importer plugins can use write_post for both single and non-single file
> output? Now all importer plugins which want to support both ways have to do
> such a rather complex if-then-else statement.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/getnikola/nikola/pull/2292/files#r55441307>.
>